server/ai_live_video.go

```diff
@@ -199,30 +198,47 @@ func startEventsSubscribe(ctx context.Context, url *url.URL, params aiRequestPar
 				continue
 			}
 
-			stream := params.liveParams.stream
-
-			if stream == "" {
-				clog.Infof(ctx, "Stream ID is missing")
-				continue
-			}
-
-			var status monitor.PipelineStatus
+			var status map[string]interface{}
 			if err := json.Unmarshal(body, &status); err != nil {
 				clog.Infof(ctx, "Failed to parse JSON from events subscription: %s", err)
 				continue
 			}
 
-			status.StreamID = &stream
+			if _, ok := status["stream_id"]; !ok {
+				clog.Infof(ctx, "Received event for stream=%s status=%+v", stream, status)
+				status["stream_id"] = stream
+				status["request_id"] = params.liveParams.requestID
+			}
 
-			// TODO: update the in-memory pipeline status
-			// monitor.UpdatePipelineStatus(stream, status)
+			monitor.SendQueueEventAsync("stream_status", status)
 
-			clog.Infof(ctx, "Received event for stream=%s status=%+v", stream, status)
+			streamStatus := map[string]interface{}{
+				"updatedAt": time.Now(),
+			}
 
-			monitor.SendQueueEventAsync(
-				"stream_status",
-				status,
-			)
+			lastStreamStatus, ok := monitor.GetStreamStatus(stream)
+			if !ok {
+				lastStreamStatus = make(map[string]interface{})
+			}
+
+			if typeVal, ok := status["type"]; ok && typeVal == "status" {
+				if logs, ok := status["last_restart_logs"]; !ok || logs == nil {
+					if lastLogs, exists := lastStreamStatus["lastRestartLogs"]; exists && lastLogs != nil {
+						streamStatus["lastRestartLogs"] = lastLogs
+					}
+				} else {
+					streamStatus["lastRestartLogs"] = logs
+				}
```

Comment:
I think this is a bit overcomplicated. This will work almost [1] the same:
```suggestion
				if logs, ok := status["last_restart_logs"]; !ok || logs == nil {
				    streamStatus["last_restart_logs"] = lastStreamStatus["last_restart_logs"]
				}
```

Notice that: 
- you had the wrong casing for the field name. I miss typing to catch stuff like this but let's stay with this untyped map 🥲
- you don't need to set `streamStatus["lastRestartLogs"] = logs`, that's a noop

[1] it will always add a `null` `last_restart_logs`, instead of omitting it when it was never sent. But I think it's fine?

Reply:
In general, seems like you're using `camelCase`. You should not since the events come from python which is `snake_case`. This might be inconsistent with other APIs (not sure), but let's keep it as snake for now. Let's say all new AI APIs should be snake.